### PR TITLE
feat: add Google login flow and inbox creation 

### PR DESCRIPTION
**Pull request by scmmishra**
This PR adds the following changes

1. Refactor `microsoft/callbacks_controller` to move common logic to `oauth_callback_controller`, most of the logic is re-used for Google
2. Add UI components, `googleClient` and I18n entries for Google login
3. Add Google callback and inbox creation
4. Add a `joinUrl` utility along with specs (need to move it to utils)
5. Add `GoogleConcern`, `Google::AuthorizationsController` and `Google::CallbacksController`

> Note: The UI is hidden for now, so we can merge this without any hiccups, to enable it just revert the commit `05c18de`

### Preview

https://github.com/chatwoot/chatwoot/assets/18097732/1606d150-4561-49dc-838d-e0b00fe49ce3

### Linear Tickers

[CW-3370](https://linear.app/chatwoot/issue/CW-3370)
[CW-3371](https://linear.app/chatwoot/issue/CW-3371)

### CODE REVIEW

**Comment by scmmishra** [comment]
@muhsin-k can you review and test this? Revert the `05c18de` commit and it will enable the Google flow

File: app/javascript/shared/helpers/joinUrl.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Join multiple paths together with a base URL
+ * NOTE: This function is not designed to handle query strings or fragments
+ *
+ * @param {string} baseUrl - The base URL to join the paths to
+ * @param  {...string} paths - The paths to join
+ * @returns {string} - The full URL
+ */
+export function joinUrl(baseUrl, ...paths) {
```
Comment (muhsin-k) [inline app/javascript/shared/helpers/joinUrl.js:18]: @scmmishra Are we using this method anywhere?

**Review by muhsin-k — APPROVED** [review]

File: app/javascript/shared/helpers/joinUrl.js (hunk 1 above)
Comment (scmmishra) [reply app/javascript/shared/helpers/joinUrl.js]: Nah, was used earlier, removed it

**Comment by scmmishra** [comment]
@pranavrajs do you wanna take a peek at the code? This is tested on staging

**Pull request merged**